src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -22,6 +22,58 @@ var ReactServerRendering;
 var ID_ATTRIBUTE_NAME;
 var ROOT_ATTRIBUTE_NAME;
 
+// Renders text using SSR and then stuffs it into a DOM node, which is returned.
+// Does not perform client-side reconnect.
+function renderOnServer(reactElement) {
+  const markup = ReactServerRendering.renderToString(
+		reactElement
+	);
+
+  var domElement = document.createElement('div');
+  domElement.innerHTML = markup;
+  return domElement;
+}
+
+function renderOnClient(reactElement, domElement, shouldMatch = true) {
+  // we don't use spyOn(console, 'error') here because we want to be able to call this
+  // function multiple times, and spyOn throws an error if you give it a function that is
+  // already being spied upon.
+  var oldConsoleError = console.error;
+  console.error = jasmine.createSpy();
+  ReactDOM.render(reactElement, domElement);
+  if (shouldMatch && console.error.argsForCall.length !== 0) {
+    // in Jasmine >= 2.3, we would just use fail() here.
+    console.log(`An unexpected error was logged during markup reconnection: '${console.error.argsForCall[0][0]}'`);
+    expect(`${console.error.argsForCall.length} errors logged.`).toBe('No errors logged.');
+  } else if (!shouldMatch && console.error.argsForCall.length === 0) {
+    console.log('Expected an error during reconnection but no error was logged.');
+    expect(`${console.error.argsForCall.length} errors logged.`).toBe('1 error logged.');
+  }
+  console.error = oldConsoleError;
+  return domElement;
+}
+// renders the first element with renderToString, puts it into a DOM node,
+// runs React.render on that DOM node with the second element. returns the DOM
+// node.
+function connectToServerRendering(
+  elementToRenderOnServer,
+  elementToRenderOnClient = elementToRenderOnServer,
+  shouldMatch = true
+) {
+  return renderOnClient(
+    elementToRenderOnClient,
+    renderOnServer(elementToRenderOnServer),
+    shouldMatch);
+}
+
+function expectMarkupMismatch(elementToRenderOnServer, elementToRenderOnClient = elementToRenderOnServer) {
```

Comment:
That's fair. I was just copying the signature around, and it makes sense in the other function signatures, where `shouldMatch` is also an argument. 

Thanks!